### PR TITLE
Allow parameter modification from an input file for restarts

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

What is says on the label quality of life improvement.
Our use case is that we use Parthenon for post-processing and thus need to pass (significant amounts) of new parameters, e.g., for histograms which gets hard to manage (and some even hit a limit in terms of what the bash allows).

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]
This looks fine, but we should probably check this machinery all works nicely somehow... maybe with the restart regression test?

File: doc/sphinx/src/README.rst [hunk 1]
```diff
@@ -306,7 +306,12 @@ command. For example, the ``refine_tol`` parameter in the
 appending ``parthenon/refinement0/refine_tol=my_new_value`` to the
 launch command (e.g.,
 ``srun ./myapp -i my_input.file parthenon/refinement0/refine_tol=my_new_value``).
-This similarly applies to simulations that are restarted.
+This similarly applies to simulations that are restarted, where modifications
+from the command line and through an input file are possible, e.g.,
+``srun ./myapp -r out0.rhdf -i modified_input.in parthenon/refinement0/refine_tol=my_new_value``.
+In the latter case, the input stored in the restart file will be read first,
+then updated from the content in the input file, and finalyl modified from the
```
Comment (lroberts36) [inline doc/sphinx/src/README.rst]:
```suggestion
then updated from the content in the input file, and finaly modified from the
```

File: doc/sphinx/src/outputs.rst [hunk 2]
```diff
@@ -134,8 +134,9 @@ This will produce an hdf5 (``.rhdf``) output file every 1 units of
 simulation time that can be used for restarting the simulation.
 
 To use this restart file, simply specify the restart file with a
-``-r <restart.rhdf>`` at the command line. It is an error to specify an
-input file with the ``-i`` flag when using the restart option.
+``-r <restart.rhdf>`` at the command line. If both ``-r <restart.rhdf>``
+and ``-i <input.in>`` are specicifed, the simulation will be restarted from
+the restart file with input parameter updated (or added) from the input file.
```
Comment (lroberts36) [inline doc/sphinx/src/outputs.rst]:
```suggestion
the restart file with input parameters updated (or added) from the input file.
```

**Review by lroberts36 — APPROVED** [review]
LGTM

File: doc/sphinx/src/README.rst (hunk 1 above)
Comment (pdmullen) [reply doc/sphinx/src/README.rst]: finally

**Review by pdmullen — APPROVED** [review]
Should we warn the user if CLI *and* an input file try to override the same parameter in `ParameterInput`?

**Comment by pgrete** [comment]
> Should we warn the user if CLI _and_ an input file try to override the same parameter in `ParameterInput`?

As discussed during the last sync, this would require quite a bit of additional machinery so we'll not warn (for now).

**Comment by pgrete** [comment]
> This looks fine, but we should probably check this machinery all works nicely somehow... maybe with the restart regression test?

done

**Pull request merged**